src/escape.d

```diff
@@ -92,20 +92,9 @@ bool checkEscape(Scope* sc, Expression e, bool gag)
                 Type tb = v.type.toBasetype();
                 if (v.isScope())
                 {
-                    /* Today, scope attribute almost doesn't work for escape analysis.
-                     * Until the semantics will be completed, it should be left as-is.
-                     * See also: fail_compilation/fail_scope.d
-                     */
-                    if (tb.ty == Tarray || tb.ty == Tsarray || tb.ty == Tclass || tb.ty == Tdelegate)
-                    {
-                        if (v.needsScopeDtor() || tb.ty == Tclass)
-                        {
-                            error(e.loc, "escaping reference to scope local %s", v);
-                            return;
-                        }
-                    }
+                    error(e.loc, "scope variable %s may not be returned", v);
```

Comment:
"may not" does not mean "optional". I also prefer to avoid contractions in code because grepping for them is a pain because of the quote.